pages/your_tasks/virtual_dev_environments.md

```diff
@@ -0,0 +1,120 @@
+---
+title: Reproducible virtual software development environments
+description: Virtual software development environments for reproducible research 
+contributors: ["Aleksandra Nenadic"]
+page_id: virtual_environments
+related_pages: []
+---
+
+## What are virtual software development environments?
+
+A virtual software development environment helps us create an **isolated working copy** of a software project that uses a specific 
+version of a programming language interpreter/compiler (e.g. Python 3.11) together with specific versions of a number of external 
+libraries (dependencies) required by our software installed into that virtual environment. 
+
+Virtual environments are typically implemented as directories with a particular structure (usually contained within software projects 
+but can be outside too), containing links to specified dependencies allowing isolation from other software projects on your machine 
+that may require different versions of the same programming language or external libraries.
+
+## Why should you use virtual software development environments? 
+
+### Description 
+
+Software applications often rely on external libraries that you need to install and manage on your system as you develop your software. 
+Software will sometimes need a specific version of an external library (e.g. because they were written to work with feature, class, or 
+function that may have been updated in more recent versions), or a specific version of the program language interpreter/compiler 
+(e.g. consider legacy software requiring Python 2 vs. new applications written in Python 3). 
+
+This means that each software application you develop on your machine may require a different setup and a set of dependencies so it is useful to be 
+able to keep these configurations separate to avoid confusion between projects. 
+The solution for this problem is to create a self-contained virtual environment per project, which contains a particular version of your 
+programming language interpreter/compiler plus a number of additional external libraries.
+
+Another big motivator for using virtual environments is that they make **sharing your code with others** (users or developers) much easier.
+By sharing a description of your virtual development environment you enable others to quickly replicate the same environment 
+on their machines and run or further develop your software - making your work **portable**, **reusable** and more **reproducible**.
+
+### Considerations
+
+- As more external libraries are added to your software project over time, you can add them to its specific virtual environment
```

Comment:
Is it worth linking out to this as a bad example of a spaghetti setup when things are not isolated - https://xkcd.com/1987/

Reply:
Fixed via [4457d22](https://github.com/EVERSE-ResearchSoftware/RSQKit/commit/4457d22ad92f2fccc4274076b0e01d5a8f192f12).